Resolve unaligned pointer value warning (Macintosh build)

The TravisCI Macintosh gcc and clang builds both report the warning below:

```
Packet.cpp:13:16: warning: taking address of packed member 'checksum' of class or structure 'PacketHeader' may result in an unaligned pointer value [-Waddress-of-packed-member]
        int* data = (&header.checksum) + 1;
                      ^~~~~~~~~~~~~~~
```

I'm not actually sure how to resolve this warning.

--- a/server/Packet.cpp
+++ b/server/Packet.cpp
@@ -10,7 +10,7 @@ int Packet::Checksum()
 
 	// Checksum the post-checksum fields
 	int dataChecksum = 0;
-	int* data = (&header.checksum) + 1;
+	auto* data = reinterpret_cast<int*>(reinterpret_cast<char*>(&header.checksum) + 4);
 
 	// Handle DWORD at a time
 	for (int i = header.sizeOfPayload / 4; i > 0; --i)

